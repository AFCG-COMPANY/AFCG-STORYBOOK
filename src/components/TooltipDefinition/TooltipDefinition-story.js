import React from 'react';
import { storiesOf } from '@storybook/react';
import TooltipDefinition from '../TooltipDefinition';

storiesOf('TooltipDefinition', module)
  .addWithInfo(
    'default',
    `
    Definition Tooltip
    `,
    () => (
      <div style={{ marginTop: '2rem' }}>
        <TooltipDefinition tooltipText="Brief description of the dotted, underlined word above.">
          Definition Tooltip
        </TooltipDefinition>
      </div>
    )
  )
  .addWithInfo(
    'with custom direction',
    `
    Definition Tooltip with custom direction
    `,
    () => (
      <div style={{ marginTop: '2rem' }}>
        <TooltipDefinition
          direction="top"
          tooltipText="Brief description of the dotted, underlined word above.">
          Definition Tooltip
        </TooltipDefinition>
      </div>
    )
  );
