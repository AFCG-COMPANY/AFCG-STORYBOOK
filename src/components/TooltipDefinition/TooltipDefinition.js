import cx from 'classnames';
import React from 'react';
import PropTypes from 'prop-types';
import setupGetInstanceId from '../../tools/setupGetInstanceId';

const getInstanceId = setupGetInstanceId();

const TooltipDefinition = ({
  id,
  className,
  children,
  direction,
  tooltipText,
  ...rest
}) => {
  const tooltipId = id || `definition-tooltip-${getInstanceId()}`;
  const definitionClassName = cx({
    [className]: !!className,
    'bx--tooltip--definition': true,
  });
  const directionClassName = cx({
    'bx--tooltip--definition__bottom': direction === 'bottom',
    'bx--tooltip--definition__top': direction === 'top',
  });
  return (
    <div {...rest} className={definitionClassName}>
      <button className="bx--tooltip__trigger" aria-describedby={tooltipId}>
        {children}
      </button>
      <div
        id={tooltipId}
        className={directionClassName}
        role="tooltip"
        aria-label={tooltipText}>
        <span className="bx--tooltip__caret" />
        <p>{tooltipText}</p>
      </div>
    </div>
  );
};

TooltipDefinition.propTypes = {
  /**
   * Specify the tooltip trigger text that is rendered to the UI for the user to
   * interact with in order to display the tooltip.
   */
  children: PropTypes.string.isRequired,

  /**
   * Specify the direction of the tooltip. Can be either bottom or top.
   */
  direction: PropTypes.oneOf(['top', 'bottom']).isRequired,

  /**
   * Optionally specify a custom id for the tooltip. If one is not provided, we
   * generate a unique id for you.
   */
  id: PropTypes.string,

  /**
   * Provide the text that will be displayed in the tooltip when it is rendered.
   */
  tooltipText: PropTypes.node.isRequired,
};

TooltipDefinition.defaultProps = {
  direction: 'bottom',
};

export default TooltipDefinition;
