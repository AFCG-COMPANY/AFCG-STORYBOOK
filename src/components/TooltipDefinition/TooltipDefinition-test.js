import React from 'react';
import { mount } from 'enzyme';
import TooltipDefinition from '../TooltipDefinition';

describe('TooltipDefinition', () => {
  let mockProps;

  beforeEach(() => {
    mockProps = {
      direction: 'bottom',
      children: 'tooltip trigger',
      className: 'custom-class',
      tooltipText: 'tooltip text',
    };
  });

  it('should render', () => {
    const wrapper = mount(<TooltipDefinition {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow the user to specify the direction', () => {
    const wrapper = mount(<TooltipDefinition {...mockProps} direction="top" />);
    expect(wrapper).toMatchSnapshot();
  });
});
