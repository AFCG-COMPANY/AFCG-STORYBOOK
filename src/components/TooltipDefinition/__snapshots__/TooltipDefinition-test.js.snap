// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipDefinition should allow the user to specify the direction 1`] = `
<TooltipDefinition
  className="custom-class"
  direction="top"
  tooltipText="tooltip text"
>
  <div
    className="custom-class bx--tooltip--definition"
  >
    <button
      aria-describedby="definition-tooltip-2"
      className="bx--tooltip__trigger"
    >
      tooltip trigger
    </button>
    <div
      aria-label="tooltip text"
      className="bx--tooltip--definition__top"
      id="definition-tooltip-2"
      role="tooltip"
    >
      <span
        className="bx--tooltip__caret"
      />
      <p>
        tooltip text
      </p>
    </div>
  </div>
</TooltipDefinition>
`;

exports[`TooltipDefinition should render 1`] = `
<TooltipDefinition
  className="custom-class"
  direction="bottom"
  tooltipText="tooltip text"
>
  <div
    className="custom-class bx--tooltip--definition"
  >
    <button
      aria-describedby="definition-tooltip-1"
      className="bx--tooltip__trigger"
    >
      tooltip trigger
    </button>
    <div
      aria-label="tooltip text"
      className="bx--tooltip--definition__bottom"
      id="definition-tooltip-1"
      role="tooltip"
    >
      <span
        className="bx--tooltip__caret"
      />
      <p>
        tooltip text
      </p>
    </div>
  </div>
</TooltipDefinition>
`;
